libcudacxx/include/cuda/std/__functional/invoke.h

```diff
@@ -541,6 +541,10 @@ invoke(_Fn&& __f, _Args&&... __args) noexcept(is_nothrow_invocable_v<_Fn, _Args.
 
 #endif // _CCCL_STD_VER > 2011
 
+/// The type of intermediate accumulator (according to P2322R6)
+template <typename Invokable, typename InitT, typename InputT>
+using __accumulator_t = typename decay<typename _CUDA_VSTD::__invoke_of<Invokable, InitT, InputT>::type>::type;
+
```

Comment:
Remark: Really awesome to see this! Please create an issue as well to replace `cub::detail::accumulator_t` by this new facility.

Also: I would have swapped the last two template parameters:
```c++
template <typename Invokable, typename InputT, typename InitT = InputT>
```
Then we could use it as well for algorithms without an initial value. But maybe there was a reason why people designed it the other way around for CUB. Anyhow, we can also change this later since it's an internal type function.